### ISSUE TITLE
.NET enums should not be automatically converted to Python int and back

### ISSUE THREAD

**Issue by lostmsu** [pythonnet/pythonnet#1220]
From https://github.com/pythonnet/pythonnet/pull/1179

In Python, it should be possible to refer to enum members: `MyEnum.Member`, and to create arbitrary enum values from `int` explicitly like this: `MyEnum(42)`. But by default a number and a value of some enum should not be freely convertible to each other. E.g. this should fail:

```csharp
void Method(MyEnum value) => ...;
```

Python:
```python
Method(42); # <- this should be a TypeError
```

We can add a generic enum codec, that could be enabled on demand to restore old behavior.

**Comment by tminka** [comment]
This is related to #1099

**Comment by wollmich** [comment]
I support that. This would solve a lot of problems, e.g.: https://stackoverflow.com/q/65805093/7556646.

**Comment by TheCollegedude** [comment]
Is there a workaround? I tried using IPyObjectDecoder but it is marked as "obsolete". The wiki, where I found it, seems to be 
 outdated.

**Comment by lostmsu** [comment]
@TheCollegedude I've just submitted a new PR for `master` branch to disable default conversion.

If you are planning to stay on 2.5, you can use `IPyObjectDecoder`. It is marked obsolete, as it might change in 3.0 (so far it has not).

**Comment by LeeDongGeon1996** [comment]
I think this should be a hot fix.

If there is C# codes like the below:
```C#
namespace myname.mynamespace
{
    public static partial class MyClass
    {
        public static IEnumerable<MyResult> MyFunc<TSometing>(
            this IEnumerable<TSomething> something,
            int num)
            where TSomething : ISomething
        {
             // do something.
        }

        public static IEnumerable<MyResult> MyFunc<TSometing>(
            this IEnumerable<TSomething> something,
            int num,
            MyEnum myenum = MyEnum.Elem)
            where TSomething : ISomething
        {
              // do something.
        }
    }
}
```

The second overloaded method can never be called:
```python
from myname.mynamespace import MyClass
from myname.mynamespace import MyEnum

#Something is sub-class of ISomething .
MyClass.MyFunc[Something](something_list, 10, MyEnum.Elem2)

>>>
TypeError: No method matches given arguments for GetSma: (<class 'System.Collections.Generic.0, Culture=neutral, PublicKeyToken=null]]'>, <class 'int'>, <class 'int'>)
```
I also tried to call the second one manually by using `Overload(__overload__)`, fetching `Type` from `System.Reflection`, it doesn't work

**Comment by lostmsu** [comment]
@LeeDongGeon1996 this has already been fixed in pythonnet 3.0 previews. We do not plan to backport it.

**Comment by LeeDongGeon1996** [comment]
Thank. But, I have a problem with using 3.0 preview.
 I'm using it on Anaconda env with Mono on Mac. Is there any way to install preview with `conda install -c conda-forge` command?
